pkg/ds/daemon_set.go

```diff
@@ -339,12 +362,17 @@ func (ds *daemonSet) clearPods() error {
 	toUnscheduleSorted := types.NewNodeSet(currentNodes...).ListNodes()
 	ds.logger.NoFields().Infof("Need to unschedule %d nodes", len(toUnscheduleSorted))
 
+	if len(toUnscheduleSorted) > 0 {
+		ds.CancelReplication()
```

Comment:
you can call this unconditionally here, the daemon set is being deleted and `CancelReplication` protects itself against there not being a replication in progress